In [2]:
import pandas as pd

In [36]:
df = pd.read_csv("train.csv")
pd.set_option("display.max_rows", 1000)

In [4]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [59]:
print('COLUNAS:')
df.columns
for i in df.columns:
    print(i)

COLUNAS:
PassengerId
Survived
Pclass
Name
Sex
Age
SibSp
Parch
Ticket
Fare
Cabin
Embarked


In [60]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


****QUANTIDADE DE PASSAGEIROS?****

In [ ]:
qtd = df['Name'].count()
print(f'TOTAL PASSAGEIROS: {qtd}')

TOTAL PASSAGEIROS: 891


TAXA DE SOBREVIVENCIA

In [76]:
import pandas as pd
import matplotlib.pyplot as plt

total = df['Name'].count()
sobrev = df[df['Survived'] == 1].shape[0]
mortos = total - sobrev
taxa = round((sobrev / total) * 100)

print(f'Total de Passageiros: {total}')
print(f'Total de Sobreviventes: {sobrev}')
print(f'Taxa de sobrevivência: {taxa}%')

Total de Passageiros: 891
Total de Sobreviventes: 342
Taxa de sobrevivência: 38%


****QUANTIDADE POR SEXO?****

In [77]:
tot = df['Sex'].count()
masculino = df.query('Sex == "male"').shape[0] 
feminino = df.query('Sex == "female"').shape[0]

taxam = round(((masculino / tot) * 100), 2)
taxaf = round(((feminino / tot) * 100), 2)

# Exibição
print(f'PASSAGEIROS: {tot}')
print('--'*15)
print(f'Masculino: {masculino}')
print(f'{taxam}%')
print(f'Feminino: {feminino}')
print(f'{taxaf}%')

PASSAGEIROS: 891
------------------------------
Masculino: 577
64.76%
Feminino: 314
35.24%


****ANÁLISES POR IDADE****

In [127]:
print(f'ANÁLISE POR IDADE')
media_idade = round(df['Age'].mean(), 2)
mediana = df['Age'].median()
maior = df['Age'].max()
menor = df['Age'].min()
homens = df[df['Sex'] == 'male']['Age'].mean()
mulheres = df[df['Sex'] == 'female']['Age'].mean()

print(f'Média: {media_idade}')
print(f'Mediana: {mediana}')
print(f'Maior Idade: {maior}')
print(f'Menor Idade: {menor}  ~ 5 meses')
print()
print(f'Média de Idade dos Homens: {round(homens, 2)}')
print(f'Média de Idade das Mulheres: {round(mulheres, 2)}')

ANÁLISE POR IDADE
Média: 29.7
Mediana: 28.0
Maior Idade: 80.0
Menor Idade: 0.42  ~ 5 meses

Média de Idade dos Homens: 30.73
Média de Idade das Mulheres: 27.92


****SOBREVIVENTES X ÓBITOS****

In [82]:
soma_sobreviventes = df.groupby('Age')['Survived'].sum()
soma_sobreviventes_acima_zero = soma_sobreviventes[soma_sobreviventes > 0].sort_values(ascending=False)
print('RANKING DE MORTES POR IDADE')
soma_sobreviventes_acima_zero.head(22)

RANKING DE MORTES POR IDADE


Age
24.0    15
36.0    11
22.0    11
27.0    11
35.0    11
30.0    10
32.0     9
18.0     9
19.0     9
31.0     8
29.0     8
28.0     7
4.0      7
40.0     6
17.0     6
25.0     6
26.0     6
48.0     6
33.0     6
34.0     6
16.0     6
42.0     6
Name: Survived, dtype: int64

In [83]:
soma_sobreviventes = df.groupby('Age')['Survived'].sum()
soma_sobreviventes_acima_zero = soma_sobreviventes[soma_sobreviventes == 0].sort_values(ascending=False)
print('IDADES SEM MORTES')
soma_sobreviventes_acima_zero.head(45)


IDADES SEM MORTES


Age
10.0    0
14.5    0
20.5    0
23.5    0
24.5    0
28.5    0
30.5    0
34.5    0
36.5    0
40.5    0
45.5    0
46.0    0
55.5    0
57.0    0
59.0    0
61.0    0
64.0    0
65.0    0
66.0    0
70.0    0
70.5    0
71.0    0
74.0    0
Name: Survived, dtype: int64

In [86]:
print( 'TOTAL DE PASSAGEIROS POR CLASSE')
df.groupby('Pclass')['Pclass'].count()

TOTAL DE PASSAGEIROS POR CLASSE


Pclass
1    216
2    184
3    491
Name: Pclass, dtype: int64

In [85]:
print('MORTES POR CLASSE')
nao_sobreviventes = df[df['Survived'] == 0]

soma_nao_sobreviventes_por_classe = nao_sobreviventes.groupby('Pclass')['Survived'].count()

soma_nao_sobreviventes_por_classe


MORTES POR CLASSE


Pclass
1     80
2     97
3    372
Name: Survived, dtype: int64

In [87]:
print('TAXA DE SOBREVIVÊNCIA POR CLASSE')

tot = df[df['Survived'] == 1]['Pclass'].count()

totclass1 = df[df['Pclass'] == 1].shape[0]
class1 = df[(df['Pclass'] == 1) & (df['Survived'] == 1)].shape[0]

totclass2 = df[df['Pclass'] == 2].shape[0]
class2 = df[(df['Pclass'] == 2) & (df['Survived'] == 1)].shape[0]

totclass3 = df[df['Pclass'] == 3].shape[0]
class3 = df[(df['Pclass'] == 3) & (df['Survived'] == 1)].shape[0]

taxa1 = round((class1 / totclass1) * 100, 2)
taxa2 = round((class2 / totclass2) * 100, 2)
taxa3 = round((class3 / totclass3) * 100, 2)  # Corrigido aqui

print('--' * 15)
print(f'Qtd de Sobreviventes: {tot}')
print('--' * 15)
print(f'Classe 1: {class1} de {totclass1} -- {taxa1}%')
print(f'Classe 2: {class2} de {totclass2} -- {taxa2}%')
print(f'Classe 3: {class3} de {totclass3} -- {taxa3}%')

TAXA DE SOBREVIVÊNCIA POR CLASSE
------------------------------
Qtd de Sobreviventes: 342
------------------------------
Classe 1: 136 de 216 -- 62.96%
Classe 2: 87 de 184 -- 47.28%
Classe 3: 119 de 491 -- 24.24%


CLASSES SOBREVIVENTES

CLASSE 01

In [88]:
# Dados da 1ª classe já calculados
total1 = df[(df['Pclass'] == 1) & (df['Survived'] == 1)].shape[0]
total = df[df['Pclass'] == 1].shape[0]
qtd_m = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)].shape[0]
qtd_h = df[(df['Sex'] == 'male') & (df['Pclass'] == 1)].shape[0]
mulheres_1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1) & (df['Survived'] == 1)].shape[0]
homens_1 = df[(df['Sex'] == 'male') & (df['Pclass'] == 1) & (df['Survived'] == 1)].shape[0]
taxam = round(((mulheres_1 / total1) * 100), 2)
taxah = round(((homens_1 / total1) * 100), 2)

# Impressão
print('CLASSE 1')
print(f'Total: {total}')
print(f'QTD Mulheres: {qtd_m}')
print(f'QTD Homens: {qtd_h}')
print('--'*15)
print(f'Sobreviventes: {total1}')
print(f'Mulheres: {mulheres_1} -- {taxam}%')
print(f'Homens: {homens_1} -- {taxah}%')
print('--'*15)

CLASSE 1
Total: 216
QTD Mulheres: 94
QTD Homens: 122
------------------------------
Sobreviventes: 136
Mulheres: 91 -- 66.91%
Homens: 45 -- 33.09%
------------------------------


In [71]:
m_idade = round((df[(df['Sex'] == 'female') & (df['Survived'] == 1) & (df['Pclass'] == 1)]['Age'].mean()),2)

print(f'Média de idade do sexo feminino sobreviventes da classe 1:\n{m_idade}')



Média de idade do sexo feminino sobreviventes da classe 1:
34.94


CLASSE 02

In [ ]:
# Dados já calculados para 3ª classe
total1 = df[(df['Pclass'] == 3) & (df['Survived'] == 1)].shape[0]
total = df[df['Pclass'] == 3].shape[0]
qtd_m = df[(df['Sex'] == 'female') & (df['Pclass'] == 3)].shape[0]
qtd_h = df[(df['Sex'] == 'male') & (df['Pclass'] == 3)].shape[0]
mulheres_1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 3) & (df['Survived'] == 1)].shape[0]
homens_1 = df[(df['Sex'] == 'male') & (df['Pclass'] == 3) & (df['Survived'] == 1)].shape[0]
taxam = round(((mulheres_1 / total1) * 100), 2)
taxah = round(((homens_1 / total1) * 100), 2)

# Impressão dos dados
print('CLASSE 2')
print(f'Total: {total}')
print(f'QTD Mulheres: {qtd_m}')
print(f'QTD Homens: {qtd_h}')
print('--'*15)
print(f'Sobreviventes: {total1}')
print(f'Mulheres: {mulheres_1} -- {taxam}%')
print(f'Homens: {homens_1} -- {taxah}%')
print('--'*15)

CLASSE 3
Total: 491
QTD Mulheres: 144
QTD Homens: 347
------------------------------
Sobreviventes: 119
Mulheres: 72 -- 60.5%
Homens: 47 -- 39.5%
------------------------------


CLASSE 03

In [90]:
# Dados da 3ª classe
total1 = df[(df['Pclass'] == 3) & (df['Survived'] == 1)].shape[0]
total = df[df['Pclass'] == 3].shape[0]
qtd_m = df[(df['Sex'] == 'female') & (df['Pclass'] == 3)].shape[0]
qtd_h = df[(df['Sex'] == 'male') & (df['Pclass'] == 3)].shape[0]
mulheres_1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 3) & (df['Survived'] == 1)].shape[0]
homens_1 = df[(df['Sex'] == 'male') & (df['Pclass'] == 3) & (df['Survived'] == 1)].shape[0]
taxam = round(((mulheres_1 / total1) * 100), 2)
taxah = round(((homens_1 / total1) * 100), 2)

print('CLASSE 3')
print(f'Total: {total}')
print(f'QTD Mulheres: {qtd_m}')
print(f'QTD Homens: {qtd_h}')
print('--'*15)
print(f'Sobreviventes: {total1}')
print(f'Mulheres: {mulheres_1} -- {taxam}%')
print(f'Homens: {homens_1} -- {taxah}%')
print('--'*15)

CLASSE 3
Total: 491
QTD Mulheres: 144
QTD Homens: 347
------------------------------
Sobreviventes: 119
Mulheres: 72 -- 60.5%
Homens: 47 -- 39.5%
------------------------------


In [55]:
m_idade = round((df[(df['Sex'] == 'male') & (df['Survived'] == 0) & (df['Pclass'] == 3)]['Age'].mean()),2)

print(f'Média de idade do sexo masculino morto da classe 3:\n{m_idade}')

Média de idade do sexo masculino morto da classe 3:
27.26


**ANÁLISE DAS CABINES**

In [56]:
qtd_cabin = df['Cabin'].nunique()
print(f'No Titanic tinham: {qtd_cabin} Cabines')

No Titanic tinham: 147 Cabines


In [29]:
print('CABINES COM MAIS PESSOAS:')
df.groupby('Cabin')['Cabin'].count().sort_values(ascending= False).head(25)


CABINES COM MAIS PESSOAS:


Cabin
G6                 4
C23 C25 C27        4
B96 B98            4
F2                 3
D                  3
E101               3
F33                3
C22 C26            3
B57 B59 B63 B66    2
B77                2
B58 B60            2
B22                2
B28                2
B49                2
B51 B53 B55        2
B20                2
B18                2
C123               2
C65                2
D35                2
D33                2
D26                2
D20                2
E44                2
E121               2
Name: Cabin, dtype: int64

In [91]:
print('MORTES POR CABINE:')
nao_sobreviventes = df[df['Survived'] == 0]

contagem_nao_sobreviventes = nao_sobreviventes.groupby('Cabin')['Cabin'].count().sort_values(ascending=False).head(25)

print(contagem_nao_sobreviventes)


MORTES POR CABINE:
Cabin
C23 C25 C27    2
C22 C26        2
C124           2
F G73          2
G6             2
D26            2
A5             1
A7             1
B102           1
B19            1
B22            1
A19            1
A10            1
A14            1
B38            1
B37            1
B30            1
B51 B53 B55    1
B86            1
B58 B60        1
B71            1
B82 B84        1
C111           1
C110           1
B94            1
Name: Cabin, dtype: int64


In [57]:
cabines_sem_mortes = df.groupby('Cabin')['Survived'].all()
qtd_cabin = df['Cabin'].nunique()

cabines_sem_mortes = cabines_sem_mortes[cabines_sem_mortes].index

qtd_cabines_sem_mortes = len(cabines_sem_mortes)
taxa = round(((qtd_cabines_sem_mortes / qtd_cabin) * 100), 2)

print(f'Quantidade de cabines sem mortes: {qtd_cabines_sem_mortes}')
print(f'Taxa: {taxa}%')


Quantidade de cabines sem mortes: 85
Taxa: 57.82%


RESUMO
CLASSE 1 com mais sobreviventes 62.96% dessa classe sobreviverem
na mesma classe o sexo feminino foi superior, com 66.91% sobreviventes
a média de idade entre as mulheres foi: 34.94

MAIORES CHANCES DE SOBREVIVENCIA:
CLASSE: 01
SEXO: FEMININO
IDADE: 34.94

CHANCES DE MORTE ELEVADA:
CLASSE: 03
SEXO: MASCULINO
IDADE: 27.26

**EMBARCAÇÃO**

In [92]:
totc = df[df['Embarked'] == 'C'].shape[0]
totq = df[df['Embarked'] == 'Q'].shape[0]
tots = df[df['Embarked'] == 'S'].shape[0]

c = df[(df['Embarked'] == 'C') & (df['Survived'] == 1)].shape[0]
q = df[(df['Embarked'] == 'Q') & (df['Survived'] == 1)].shape[0]
s = df[(df['Embarked'] == 'S') & (df['Survived'] == 1)].shape[0]

taxac = round(((c / totc) * 100), 2)
taxaq = round(((q / totq) * 100), 2)
taxas = round(((s / tots) * 100), 2)

print('SOBREVIVENTES POR EMBARCAÇÃO:')
print('--'*20)
print(f'C Sobreviventes: {c} de {totc} -- {taxac}%')
print(f'Q Sobreviventes: {q} de {totq} -- {taxaq}%')
print(f'S Sobreviventes: {s} de {tots} -- {taxas}%')

SOBREVIVENTES POR EMBARCAÇÃO:
----------------------------------------
C Sobreviventes: 93 de 168 -- 55.36%
Q Sobreviventes: 30 de 77 -- 38.96%
S Sobreviventes: 217 de 644 -- 33.7%


In [126]:
totc = df[(df['Embarked'] == 'C') & df['Survived'] == 1].shape[0]

# TOT POR CLASSES 
totclass1 = df[(df['Embarked'] == 'C') & (df['Pclass'] == 1) & df['Survived'] == 1].shape[0]
totclass2 = df[(df['Embarked'] == 'C') & (df['Pclass'] == 2) & df['Survived'] == 1].shape[0]
totclass3 = df[(df['Embarked'] == 'C') & (df['Pclass'] == 3) & df['Survived'] == 1].shape[0]

totc_valor = round(df[(df['Embarked'] == 'C') & df['Survived'] == 1]['Fare'].mean(), 2)
totq_valor = round(df[(df['Embarked'] == 'Q') & df['Survived'] == 1]['Fare'].mean(), 2)
tots_valor = round(df[(df['Embarked'] == 'S') & df['Survived'] == 1]['Fare'].mean(), 2)

#PORCENTAGENS
class1 = round((totclass1 * 100 / totc), 2) 
class2 = round((totclass2 * 100 / totc), 2)  
class3 = round((totclass3 * 100 / totc), 2) 

print(f'Porque a Classe C teve o Maior Número de Sobreviventes: {totc}')
print(f'Sobreviventes Classe 1 embarcados na C:\n{totclass1} -- {class1}%')
print(f'Sobreviventes Classe 2 embarcados na C:\n{totclass2} -- {class2}%')
print(f'Sobreviventes Classe 3 embarcados na C:\n{totclass3} -- {class3}%')
print()
print('Valores médios das passagens dos sobreviventes em cada embarcação:')
print(f'Embacação C: {totc_valor}')
print(f'Embacação Q: {totq_valor}')
print(f'Embacação S: {tots_valor}')

Porque a Classe C teve o Maior Número de Sobreviventes: 93
Sobreviventes Classe 1 embarcados na C:
59 -- 63.44%
Sobreviventes Classe 2 embarcados na C:
9 -- 9.68%
Sobreviventes Classe 3 embarcados na C:
25 -- 26.88%

Valores médios das passagens dos sobreviventes em cada embarcação:
Embacação C: 79.72
Embacação Q: 13.18
Embacação S: 39.55


SERÁ QUE O VALOR PAGO NA PASSAGEM TEVE RELAÇÃO COM A TAXA DE SOBREVIVÊNCIA?

In [93]:
media = round(df['Fare'].median(), 2)
sobreviventes = df[df['Survived'] == 1].shape[0]

valor_1 = round(df[df['Pclass'] == 1]['Fare'].mean(), 2)
valor_2 = round(df[df['Pclass'] == 2]['Fare'].mean(), 2)
valor_3 = round(df[df['Pclass'] == 3]['Fare'].mean(), 2)

acima_media = df[(df['Fare'] > media) & (df['Survived'] == 1)].shape[0]
abaixo_media = df[(df['Fare'] < media) & (df['Survived'] == 1)].shape[0]

taxa_acima = round((acima_media / sobreviventes) * 100, 2)
taxa_abaixo = round((abaixo_media / sobreviventes) * 100, 2)

maximo = df['Fare'].max()
minimo = df['Fare'].min()

print(f'Sobreviventes: {sobreviventes}')
print('--'*25)
print(f'Qtd sobreviventes que pagaram valor acima da mediana: {acima_media} de {sobreviventes} --- {taxa_acima}%')
print(f'Qtd sobrevientes que pagaram valor abaixo da mediana: {abaixo_media} de {sobreviventes} --- {taxa_abaixo}%')
print('--'*25)
print(f'Médiana do valor das passagens: {media}')
print(f'Classe1 média paga:  {valor_1}')
print(f'Classe2 média paga:  {valor_2}')
print(f'Classe3 média paga:  {valor_3}')

print('--'*25)
print(f'Valor Máximo: {maximo}')
print(f'Valor Mínimo: {minimo}')
print('--'*25)

qtd_minimo = df[df['Fare'] <= 0].shape[0]
minimo = df[(df['Survived'] == 1) & (df['Fare'] == minimo)].shape[0]
taxa_minimo = round((minimo / qtd_minimo)* 100, 2)

qtd_maximo = df[df['Fare'] >= 227.5250].shape[0]
quinze_maximo = df[(df['Survived'] == 1) & (df['Fare'] >= 227.5250)].shape[0]
taxa_maximo = round((quinze_maximo / qtd_maximo)* 100, 2)

print(f'QTD de sobreviventes entre as pessoas que não pagaram nada:\n{minimo} de {qtd_minimo} --- {taxa_minimo}%')
print(f'QTD de sobreviventes entre as 15 pessoas que pagaram o valor máximo:\n{quinze_maximo} de {qtd_maximo} --- {taxa_maximo}%')


Sobreviventes: 342
--------------------------------------------------
Qtd sobreviventes que pagaram valor acima da mediana: 231 de 342 --- 67.54%
Qtd sobrevientes que pagaram valor abaixo da mediana: 111 de 342 --- 32.46%
--------------------------------------------------
Médiana do valor das passagens: 14.45
Classe1 média paga:  84.15
Classe2 média paga:  20.66
Classe3 média paga:  13.68
--------------------------------------------------
Valor Máximo: 512.3292
Valor Mínimo: 0.0
--------------------------------------------------
QTD de sobreviventes entre as pessoas que não pagaram nada:
1 de 15 --- 6.67%
QTD de sobreviventes entre as 15 pessoas que pagaram o valor máximo:
11 de 15 --- 73.33%


A PRIORIDADE DE SALVAMENTO NO ACIDENTE DO TITANIC 
FOI PESSOAS DA CLASSE 1, QUANTO MAIOR O VALOR PAGO MAIS CHANCES 
DE SOBREVIVER.
MULHERES DE MEIA IDADE DA CLASSE 1 TAMBÉM TIVERAM UM BOM PERCENTUAL DE SOBREVIVÊNCIA.